TypeScript build error in rc.{3-5}

# Bug report

## Describe the bug
Using TypeScript with strict mode on, this internal error comes up.

```
node_modules/@chakra-ui/system/dist/types/forward-ref.d.ts:10:186 - error TS2344: Type 'T' does not satisfy the constraint '"symbol" | "object" | "button" | "meter" | "textarea" | "style" | "progress" | "text" | "ruby" | "table" | "small" | "sub" | "circle" | "embed" | "pre" | "caption" | "menu" | "menuitem" | ... 158 more ... | (new (props: any) => Component<...>)'.
  Type 'As' is not assignable to type '"symbol" | "object" | "button" | "meter" | "textarea" | "style" | "progress" | "text" | "ruby" | "table" | "small" | "sub" | "circle" | "embed" | "pre" | "caption" | "menu" | "menuitem" | ... 158 more ... | (new (props: any) => Component<...>)'.
    Type 'string' is not assignable to type '"symbol" | "object" | "button" | "meter" | "textarea" | "style" | "progress" | "text" | "ruby" | "table" | "small" | "sub" | "circle" | "embed" | "pre" | "caption" | "menu" | "menuitem" | ... 158 more ... | (new (props: any) => Component<...>)'.
      Type 'T' is not assignable to type '"view"'.
        Type 'As' is not assignable to type '"view"'.
          Type 'string' is not assignable to type '"view"'.

10 declare type PropsOf<T extends As> = T extends React.ComponentClass<infer P> ? React.PropsWithoutRef<P> & React.RefAttributes<InstanceType<T>> : React.PropsWithRef<React.ComponentProps<T>>;
```

## Minimal reproduction

<!-- Please provide a codesandbox link or GitHub repo with a minimal
reproduction of the bug.

Official Chakra CodeSandbox templates:
- [JavaScript CodeSandbox template](https://codesandbox.io/s/github/chakra-ui/codesandbox-react-js-template/tree/master)
- [TypeScript CodeSandbox template](https://codesandbox.io/s/github/chakra-ui/codesandbox-react-ts-template/tree/master)
-->

--- a/packages/system/src/forward-ref.tsx
+++ b/packages/system/src/forward-ref.tsx
@@ -4,14 +4,13 @@
  */
 import * as React from "react"
 
-type As = string | React.ComponentType<any>
+type As = keyof JSX.IntrinsicElements | React.ComponentType<any>
 
 export type PropsWithAs<T extends As, P> = P &
   Omit<PropsOf<T>, "as" | "color" | keyof P> & {
     as?: T | As
   }
 
-// @ts-expect-error
 type PropsOf<T extends As> = React.PropsWithRef<React.ComponentProps<T>>
 
 type Merge<T, P> = P extends object ? P & Omit<T, keyof P> : T

